eclipse/src/saros/editor/EditorListener.java

```diff
@@ -201,8 +201,13 @@ protected void generateSelection() {
     if (!lastSelection.equals(selection)) {
       lastSelection = selection;
 
-      manager.generateSelection(
-          part, new saros.editor.text.TextSelection(selection.getOffset(), selection.getLength()));
+      IDocument document = viewer.getDocument();
```

Comment:
Can you please add an assertion that the document is never null ? Reading the Interface Javadoc it could. I do not know if this still holds true for textviewers.

Reply:
Will add a check logging a warning and using `EMPTY_SELECTION` instead if the document is `null`.